deeplake/core/chunk/base_chunk.py

```diff
@@ -443,13 +447,17 @@ def register_in_meta_and_headers(
            shape (Any): Parameter that shows the shape of the added elements
            update_commit_diff (bool): Parameter that shows if we need to update the commit diffs
            update_tensor_meta (bool): Parameter that shows if it is need to update tensor metas, in case of rechunk we do not need to update meta as we do not add new elements
+           num_samples (int): Number of incoming samples.
         """
-        self.register_sample_to_headers(sample_nbytes, shape)
+        self.register_sample_to_headers(sample_nbytes, shape, num_samples)
         if update_tensor_meta:
-            if self._update_tensor_meta_length:
-                self.tensor_meta.update_length(1)
-            if shape is not None:
-                self.tensor_meta.update_shape_interval(shape)
+            self.update_tensor_meta(shape, num_samples)
+
+    def update_tensor_meta(self, shape, num_samples):
+        if self._update_tensor_meta_length:
+            self.tensor_meta.update_length(num_samples)
+        if shape is not None:
+            self.tensor_meta.update_shape_interval(shape)
```

Comment:
Refactored tensor meta update to a separate method. We want to call this method alone during fast extend without registering to headers (we manipulate encoders directly there for perf boost).